app/routes/signin-oidc.js

```diff
@@ -115,6 +115,21 @@ function getRedirectPath(
   }
 }
 
+function safelyGetLoginSource(request) {
+  try {
+    return JSON.parse(
+      Buffer.from(request.query.state, "base64").toString("ascii"),
+    ).source;
+  } catch (queryStateError) {
+    request.logger.setBindings({ query: request.query });
```

Comment:
Do we need both this line and the one below? It looks like they do the same thing, sort of.